hnn_core/drives.py

```diff
@@ -10,14 +10,49 @@
                      _extract_drive_specs_from_hnn_params)
 
 
-def _get_target_populations(weights_ampa, weights_nmda):
+def _get_target_properties(weights_ampa, weights_nmda, synaptic_delays,
+                           location):
+    """Retrieve the the user-defined synaptic weight and delay parameters
+    according to their target populations cell types
+    """
+
     # allow passing weights as None, but make iterable here
     if weights_ampa is None:
         weights_ampa = dict()
     if weights_nmda is None:
         weights_nmda = dict()
-    target_populations = (set(weights_ampa.keys()) | set(weights_nmda.keys()))
-    return target_populations, weights_ampa, weights_nmda
+
+    weights_by_type = {cell_type: dict() for cell_type in
+                       (set(weights_ampa.keys()) | set(weights_ampa.keys()))}
+    for cell_type in weights_ampa:
+        weights_by_type[cell_type].update({'ampa': weights_ampa[cell_type]})
+    for cell_type in weights_nmda:
+        weights_by_type[cell_type].update({'nmda': weights_nmda[cell_type]})
+
+    target_populations = set(weights_by_type)
+    if not target_populations:
+        raise ValueError('No target cell types have been given a synaptic '
+                         'weight for this drive.')
+    # Distal drives should not target L5 basket cells according to the
+    # canonical Jones model
+    if location == 'distal' and 'L5_basket' in target_populations:
+        raise ValueError('When adding a distal drive, synaptic weight cannot '
+                         'be defined for the L5_basket cell type as this '
+                         'connection does not exist.')
```

Comment:
I thought we had some sort of check elsewhere to control for this, but I couldn't find it. Any objections?

Reply:
No objections here. I need to check but there's a few conditions that should raise a value error. Especially for the connections where the simulation errors in `network_builder.py` due to an invalid connection (target `loc` not being defined for that cell type)

This has been on my to-do for awhile though so don't worry about flushing that out here.